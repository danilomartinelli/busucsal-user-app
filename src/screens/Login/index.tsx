import React from "react";
import { StyleSheet, KeyboardAvoidingView, Button } from "react-native";
import { COLORS } from "../../resources/colors";
import { SPACER } from "../../util/consts";
import { KEYBOARD_BEHAVIOR } from "../../resources/keyboard";
import { NavigationScreenProp } from "react-navigation";

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "column",
    justifyContent: "center",
  },
  loginWrapper: {
    backgroundColor: COLORS.primary,
    justifyContent: "center",
    padding: 16,
  },
  logoContainer: {
    alignItems: "center",
  },
  logoImage: {
    width: SPACER * 18,
    height: SPACER * 18,
    resizeMode: "contain",
    justifyContent: "center",
    alignItems: "center",
  },
  formItem: {
    marginLeft: 0,
    paddingBottom: 4,
  },
  label: {
    color: "white",
  },
  btnWrapper: {
    marginTop: SPACER * 4,
  },
  buttonText: {
    color: "white",
    fontSize: 16,
  },
});

interface IProps {
  navigation?: NavigationScreenProp<{}>;
}

interface IState {
  email: string;
  password: string;
  isSubmitting: boolean;
  isDisabled: boolean;
}

class Login extends React.Component<IProps, IState> {
  constructor(props: IProps) {
    super(props);

    this.state = {
      email: "",
      password: "",
      isSubmitting: false,
      isDisabled: true,
    };
  }

  login = () => {};

  render() {
    const { isDisabled } = this.state;

    return (
      <KeyboardAvoidingView
        style={styles.container}
        behavior={KEYBOARD_BEHAVIOR}
      >
        <Button title="Logar" onPress={this.login} disabled={isDisabled} />
      </KeyboardAvoidingView>
    );
  }
}

export default Login;
