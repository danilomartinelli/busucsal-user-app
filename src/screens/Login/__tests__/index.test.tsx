import React from "react";
import { shallow } from "enzyme";
import Login from "..";
import { Button } from "react-native";

describe("Login Screen", () => {
  describe("render", () => {
    it("should render properly", () => {
      const wrapper = shallow(<Login />);
      expect(wrapper).toMatchSnapshot();
    });

    it("should render a disabled button when email is empty", () => {
      const wrapper = shallow(<Login />);

      wrapper.setState({
        email: "",
      });

      const button = wrapper.find(Button).last();
      expect(button.props().disabled).toBe(true);
    });

    it("should render a disabled button when password is empty", () => {
      const wrapper = shallow(<Login />);

      wrapper.setState({
        password: "",
      });

      const button = wrapper.find(Button).last();
      expect(button.props().disabled).toBe(true);
    });

    it("should render a enabled button when email and password is fullfiled", () => {
      const wrapper = shallow(<Login />);

      wrapper.setState({
        email: "issoae",
        password: "elemsm",
        isDisabled: false,
      });

      const button = wrapper.find(Button).last();
      expect(button.props().disabled).toBe(false);
    });
  });

  describe("behavior", () => {
    //Check if login is called
    it("should call auth service on button press", () => {
      const wrapper = shallow(<Login />);
    });
  });
});
