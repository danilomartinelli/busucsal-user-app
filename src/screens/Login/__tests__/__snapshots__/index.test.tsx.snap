// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Screen render should render properly 1`] = `ShallowWrapper {}`;
